src/include/lean/lean.h

```diff
@@ -521,6 +534,23 @@ static inline void lean_set_non_heap_header_for_big(lean_object * o, unsigned ta
 
 /* Constructor objects */
 
+/*
+Returns the number of non-scalar parameters of a given ctor-object. 
+
+For example, let `o` be a value of the following structure in Lean:
+
+```
+structure S where
+    a : String
+    b : String
+    c : Bool
+    d : UInt64
+    e : Array Nat
```

Comment:
More typical formatting is:
```suggestion
structure S where
  a : String
  b : String
  c : Bool
  d : UInt64
  e : Array Nat
```